poi-radio/src/operator/attestation.rs

```diff
@@ -641,26 +643,36 @@ pub async fn process_comparison_results(
 
     for result in result_strings {
         match result {
-            Ok(x) if x.result_type == ComparisonResultType::Match => {
-                update_divergent_subgraphs(divergent_subgraphs, &x);
-                match_strings.push(x.to_string());
-            }
-            Ok(x) if x.result_type == ComparisonResultType::NotFound => {
-                not_found_strings.push(x.to_string());
-            }
-            Ok(x) if x.result_type == ComparisonResultType::Divergent => {
-                error!(result = x.to_string(), "Found nPOI divergence!");
-                if divergent_subgraphs.contains(&x.deployment) {
-                    trace!("Known divergence, skipping notifications.");
-                } else {
-                    notifier.clone().notify(x.to_string()).await;
-                    divergent_subgraphs.insert(x.deployment.clone());
+            Ok(comparison_result) => {
+                trace!("Comparison result: {:?}", comparison_result);
+                persisted_state
+                    .handle_comparison_result(comparison_result.clone(), notifier.clone())
+                    .await;
+
+                match comparison_result.result_type {
```

Comment:
`divergent_subgraphs` can be removed overall, right? since it was only added for the previous notification state check. Maybe this entire match statement can be removed?

Reply:
yep, removed `divergent_subgraphs`, good catch! I still think this match here is nice because it handles the logging, but I just made it a bit more straightforward where it picks up the result after `handle_comparison_result` is done

Reply:
sounds good! maybe we can consider refactoring this logging code in the future 😅